lib.js

```diff
@@ -48,47 +47,51 @@ module.exports = (options) => {
             body: {
                 file_path: `public/.well-known/acme-challenge/${key}`,
                 commit_message: 'Automated Let\'s Encrypt renewal',
-                branch_name: 'master',
+                branch_name: repo.default_branch,
                 content: value
             }
         })).return([`http://${domain}/.well-known/acme-challenge/${key}`, value]);
     };
 
     const deleteChallenges = (key, repo) => {
         return Promise.resolve(gitlabRequest.delete({
-            url: `projects/${repo.id}/repository/files`,
+            url: `/projects/${repo.id}/repository/files`,
             body: {
                 file_path: `public/.well-known/acme-challenge/${key}`,
                 commit_message: 'Automated Let\'s Encrypt renewal',
-                branch_name: 'master'
+                branch_name: repo.default_branch
             }
         }));
     };
+
+    let deleteChallengesPromise = null;
+
     return Promise.join(getUrls, generateRsa(), generateRsa(), getRepository(options.repository),
         (urls, accountKp, domainKp, repo) => {
-            return LeCore.registerNewAccountAsync({
+            return ACME.registerNewAccountAsync({
                 newRegUrl: urls.newReg,
                 email: options.email,
-                accountPrivateKeyPem: accountKp.privateKeyPem,
+                accountKeypair: accountKp,
                 agreeToTerms: (tosUrl, cb) => {
                     console.log(`By using Let's Encrypt, you are agreeing to the TOS at ${tosUrl}`);
                     cb(null, true);
                 }
             }).then(() => {
-                return LeCore.getCertificateAsync({
+                return ACME.getCertificateAsync({
                     newAuthzUrl: urls.newAuthz,
                     newCertUrl: urls.newCert,
-                    domainPrivateKeyPem: domainKp.privateKeyPem,
-                    accountPrivateKeyPem: accountKp.privateKeyPem,
-                    domains: [options.domain],
+                    domainKeypair: domainKp,
+                    accountKeypair: accountKp,
+                    domains: options.domain,
                     setChallenge: (hostname, key, value, cb) => {
-                        return uploadChallenge(key, value, repo, options.domain)
+                        return Promise.resolve(deleteChallengesPromise)
+                            .then(() => uploadChallenge(key, value, repo, hostname))
                             .tap(res => console.log(`Uploaded challenge file, waiting for it to be available at ${res[0]}`))
                             .spread(pollUntilDeployed)
                             .asCallback(cb);
                     },
                     removeChallenge: (hostname, key, cb) => {
-                        return deleteChallenges(key, repo).asCallback(cb);
+                        return (deleteChallengesPromise = deleteChallenges(key, repo)).asCallback(cb);
```

Comment:
Yes, it does use separate challenge files. The reason we need to wait is becase when the commit that deletes the previous challenge file is pushed, we have to wait until the job in pipeline gets completed. If we don't do that and immediately upload the next challenge file [we get weird errors](https://github.com/rolodato/gitlab-letsencrypt/issues/13#issuecomment-296507623) (*Could not update branch master*, *Branch diverged*, etc.)